wagtail/models/__init__.py

```diff
@@ -1393,6 +1394,19 @@ class Page(AbstractPage, index.Indexed, ClusterableModel, metaclass=PageBase):
     # Privacy options for page
     private_page_options = ["password", "groups", "login"]
 
+    # Allows page types to specify a list of HTTP method names that page instances will
+    # respond to. When the request type doesn't match, Wagtail should return a response
+    # with a status code of 405.
+    allowed_http_methods = [
+        HTTPMethod.DELETE,
+        HTTPMethod.GET,
+        HTTPMethod.HEAD,
+        HTTPMethod.OPTIONS,
+        HTTPMethod.PATCH,
+        HTTPMethod.POST,
+        HTTPMethod.PUT,
+    ]
```

Comment:
Would it be better to only allow GET by default? So that it becomes an opt-in strategy for page types that need to support other methods?

Reply:
@allcaps How do you feel about my proposal above? Happy to keep things simple for now?

Reply:
Sure, lets keep it simple.

Reply:
I like this idea but agree it's best for a v7 type change maybe. Probably best to raise a new issue after this PR goes in.

Reply:
We'll need to consider form pages.